go/vt/schema/online_ddl.go

```diff
@@ -163,10 +177,30 @@ func NewOnlineDDLs(keyspace string, ddlStmt sqlparser.DDLStatement, ddlStrategyS
 				return nil, err
 			}
 		}
-		return onlineDDLs, nil
 	default:
 		return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "unsupported statement for Online DDL: %v", sqlparser.String(ddlStmt))
 	}
+
+	{
+		// SQL statement sanity checks:
+		switch ddlStmt := ddlStmt.(type) {
+		case *sqlparser.AlterTable:
+			if len(ddlStmt.AlterOptions) == 0 {
+				return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "NewOnlineDDL: cannot parse statement: %v", sqlparser.String(ddlStmt))
+			}
+		}
+
+		if !ddlStmt.IsFullyParsed() {
+			return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "NewOnlineDDL: cannot parse statement: %v", sqlparser.String(ddlStmt))
+		}
+
+		fk := &fkContraint{}
+		_ = sqlparser.Walk(fk.FkWalk, ddlStmt)
```

Comment:
nit: You can return the error from `fk.FkWalk` directly, which would also abort the traversal of the AST.

Reply:
done!